pkg/wallet/wallet_relinquish_output_test.go

```diff
@@ -0,0 +1,242 @@
+package wallet_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/fixtures"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/testabilities/testutils"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wallet/internal/testabilities"
+	"github.com/bsv-blockchain/go-sdk/transaction"
+	sdk "github.com/bsv-blockchain/go-sdk/wallet"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWalletRelinquishOutputArgsValidation(t *testing.T) {
+	errorTestCases := map[string]struct {
+		originator string
+		args       sdk.RelinquishOutputArgs
+	}{
+		"too long originator": {
+			originator: strings.Repeat("a", 251),
+			args: sdk.RelinquishOutputArgs{
+				Basket: "test-basket",
+				Output: *testutils.SdkOutpoint(t, fixtures.MockOutpoint),
+			},
+		},
+
+		"too long originator part": {
+			originator: "a." + strings.Repeat("b", 64) + ".c",
+			args: sdk.RelinquishOutputArgs{
+				Basket: "test-basket",
+				Output: *testutils.SdkOutpoint(t, fixtures.MockOutpoint),
+			},
+		},
+
+		"basket too long": {
+			originator: fixtures.DefaultOriginator,
+			args: sdk.RelinquishOutputArgs{
+				Basket: strings.Repeat("a", 301),
+				Output: *testutils.SdkOutpoint(t, fixtures.MockOutpoint),
+			},
+		},
+	}
+
+	for name, test := range errorTestCases {
+		t.Run(name, func(t *testing.T) {
+			// given:
+			given, then := testabilities.New(t)
+
+			// and:
+			aliceWallet, cleanup := given.AliceWalletWithStorage(testabilities.StorageTypeMocked)
+			defer cleanup()
+
+			// when:
+			result, err := aliceWallet.RelinquishOutput(t.Context(), test.args, test.originator)
+
+			// then:
+			then.Result(result).HasError(err)
+			then.Storage().HadNoInteraction()
+		})
+	}
+}
+
+func (s *WalletTestSuite) TestWalletRelinquishOutputErrorPaths() {
+	s.Run("output not found", func() {
+		t := s.T()
+
+		// given:
+		given := testabilities.Given(t)
+		aliceWallet, cleanup := given.AliceWalletWithStorage(s.StorageType)
+		defer cleanup()
+
+		// when:
+		args := sdk.RelinquishOutputArgs{
+			Basket: "test-basket",
+			Output: *testutils.SdkOutpoint(t, "756754d5ad8f00e05c36d89a852971c0a1dc0c10f20cd7840ead347aff475ef6:1"),
```

Comment:
Valid outpoint format is with dot `.`:
```suggestion
			Output: *testutils.SdkOutpoint(t, "756754d5ad8f00e05c36d89a852971c0a1dc0c10f20cd7840ead347aff475ef6.1"),
```

Reply:
fixed